app/modules/intelligence/provider/provider_service.py

```diff
@@ -77,6 +87,25 @@ async def set_global_ai_provider(self, user_id: str, provider: str):
         self.db.commit()
         return {"message": f"AI provider set to {provider}"}
 
+    def get_llm_provider_name(self) -> str:
+        """Returns the name of the LLM provider based on the LLM instance."""
+        llm = self.get_small_llm(agent_type=AgentType.LANGCHAIN)
+
+        if isinstance(llm, ChatOpenAI):
+            return "OpenAI"
+        elif isinstance(llm, ChatAnthropic):
+            return "Anthropic"
+        elif isinstance(llm, ChatGoogleGenerativeAI):
+            return "Google"
+        elif isinstance(llm, LLM):
+            if llm.model.split("/")[0] == "openai":
+                return "OpenAI"
+            elif llm.model.split("/")[0] == "anthropic":
+                return "Anthropic"
+            elif llm.model.split("/")[0] == "google":
+                return "Google"
```

Comment:
Adding new providers requires modifying the code and adding more if-elif blocks. Instead, we can use a map, like this:

```
model_providers = {
    "openai": "OpenAI",
    "anthropic": "Anthropic",
    "google": "Google",
}

def get_provider(llm):
    provider_key = llm.model.split("/")[0]
    return model_providers.get(provider_key, "Unknown Provider")
```

✨ This improves scalability and simplifies code updates.